### PR TITLE
Add Ubuntu and Debian VM images

### PR DESCRIPTION
**Pull request by usamasaqib**
### What does this PR do?
This PR adds debian-10, debian-11 and ubuntu-18 images for ARM64 tests.
It also add Ubuntu-16 for AMD64 tests.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: tasks/kernel_matrix_testing/vmconfig.py [hunk 1]
```diff
@@ -85,10 +87,12 @@
     "amzn_5.4": "file:///home/kernel-version-testing/rootfs/amzn2-kvm-2.0-{arch}-5.4.qcow2",
     "amzn_5.10": "file:///home/kernel-version-testing/rootfs/amzn2-kvm-2.0-{arch}-5.10.qcow2",
     "amzn_5.15": "file:///home/kernel-version-testing/rootfs/amzn2-kvm-2.0-{arch}-5.15.qcow2",
-    "fedora_35": "file:///home/kernel-version-testing/rootfs/Fedora-Cloud-Base-35.amd64.qcow2",
-    "fedora_36": "file:///home/kernel-version-testing/rootfs/Fedora-Cloud-Base-36.amd64.qcow2",
-    "fedora_37": "file:///home/kernel-version-testing/rootfs/Fedora-Cloud-Base-37.amd64.qcow2",
-    "fedora_38": "file:///home/kernel-version-testing/rootfs/Fedora-Cloud-Base-38.amd64.qcow2",
+    "fedora_35": "file:///home/kernel-version-testing/rootfs/Fedora-Cloud-Base-35.{arch}.qcow2",
+    "fedora_36": "file:///home/kernel-version-testing/rootfs/Fedora-Cloud-Base-36.{arch}.qcow2",
```
Comment (brycekahle) [inline tasks/kernel_matrix_testing/vmconfig.py:91]: we don't support fedora 35 and 36 right?

File: .gitlab/kernel_version_testing/system_probe.yml [hunk 2]
```diff
@@ -389,7 +389,7 @@ kernel_matrix_testing_run_tests_arm64:
     ARCH: "arm64"
   parallel:
     matrix:
-      - TAG: ["ubuntu_20.04", "ubuntu_22.04", "amzn_4.14", "amzn_5.4", "amzn_5.10", "fedora_37", "fedora_38"]
+      - TAG: ["ubuntu_20.04", "ubuntu_22.04", "amzn_4.14", "amzn_5.4", "amzn_5.10", "fedora_37", "fedora_38", "debian_10", "debian_11"]
```
Comment (brycekahle) [inline .gitlab/kernel_version_testing/system_probe.yml:392]: missing `ubuntu_18.04`

File: test/new-e2e/system-probe/config/vmconfig.json [hunk 3]
```diff
@@ -78,6 +78,11 @@
             "arch": "arm64",
             "console_type": "file",
             "kernels": [
+                {
+                    "dir": "bionic-server-cloudimg-arm64.qcow2",
+                    "tag": "ubuntu_18.04",
+                    "image_source": "https://dd-agent-omnibus.s3.amazonaws.com/kernel-version-testing/rootfs/bionic-server-cloudimg-arm64.qcow2.xz-test-only"
```
Comment (brycekahle) [inline test/new-e2e/system-probe/config/vmconfig.json:84]: Can you update to `main`-based images?

File: tasks/kernel_matrix_testing/vmconfig.py (hunk 1 above)
Comment (usamasaqib) [reply tasks/kernel_matrix_testing/vmconfig.py:91]: No, but this is for KMT local-dev. There is various cleanup to be performed there, which I will target in a separate PR.

File: .gitlab/kernel_version_testing/system_probe.yml (hunk 2 above)
Comment (usamasaqib) [reply .gitlab/kernel_version_testing/system_probe.yml]: Added.

**Review by brycekahle — APPROVED** [review]

**Review by brycekahle — CHANGES_REQUESTED** [review]

**Review by brycekahle — APPROVED** [review]

**Pull request merged**